modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/BaseTemplateUpgradeProcess.java

```diff
@@ -54,25 +60,32 @@ private void _upgradeDDMTemplates() throws Exception {
 					Matcher templateMatcher = getTemplatePattern().matcher(
 						script);
 
-					if (templateMatcher.find()) {
-						script = templateMatcher.replaceAll(
+					while (templateMatcher.find()) {
+						script = StringUtil.replace(
+							script, templateMatcher.group(),
 							getTemplatePatternReplacement());
 
+						if (Validator.isNotNull(getTemplateContextVariable())) {
+							script = StringUtil.replace(
+								script, templateMatcher.group(1),
```

Comment:
@velosofelipe64, taking a second look to this solution, I've realized that we're building this replacement assuming that the developer will always create the pattern with a first group associated with the variable name, however it may not be true. Can you please revisit you solution using something like:
```
script = StringUtil.replace(
	script, templateMatcher.group().substring(0, StringPool.EQUAL).trim(),
	getTemplateContextVariable());
```

Source Format will probably complain about this code due to the method chaining, several methods invoked in a single call, but the intention is just to share with you a possible way to solve the problem. :wink: 

Please start from this pull: https://github.com/velosofelipe64/liferay-portal/pull/22